src/main/java/de/tum/in/www1/artemis/service/metis/conversation/auth/ChannelAuthorizationService.java

```diff
@@ -0,0 +1,265 @@
+package de.tum.in.www1.artemis.service.metis.conversation.auth;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.metis.conversation.Channel;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.metis.ConversationParticipantRepository;
+import de.tum.in.www1.artemis.repository.metis.conversation.ChannelRepository;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+
+@Service
+public class ChannelAuthorizationService extends ConversationAuthorizationService {
+
+    private final ChannelRepository channelRepository;
+
+    public ChannelAuthorizationService(UserRepository userRepository, AuthorizationCheckService authorizationCheckService,
+            ConversationParticipantRepository conversationParticipantRepository, ChannelRepository channelRepository) {
+        super(conversationParticipantRepository, userRepository, authorizationCheckService);
+        this.channelRepository = channelRepository;
+    }
+
+    /**
+     * Checks if a user is allowed to create a channel in a course or throws an exception if not
+     *
+     * @param course the course the channel should be created in
+     * @param user   the user that wants to create the channel
+     */
+    public void isAllowedToCreateChannel(@NotNull Course course, @NotNull User user) {
+        var userToCheck = getUserIfNecessary(user);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.TEACHING_ASSISTANT, course, userToCheck);
+    }
+
+    /**
+     * Checks if a user is allowed to create a new message in a channel or throws an exception if not
+     *
+     * @param channel the channel the message should be created
+     * @param user    the user that wants to create the message
+     */
+    public void isAllowedToCreateNewAnswerPostInChannel(@NotNull Channel channel, @NotNull User user) {
+        var isArchivedChannel = !Objects.isNull(channel.getIsArchived()) && channel.getIsArchived();
+        var userToCheck = getUserIfNecessary(user);
+        var isChannelMember = isMember(channel.getId(), userToCheck.getId());
+        if (isArchivedChannel) {
```

Comment:
Maybe move this two lines above, so the method skips early instead of loading the user?

Reply:
Good idea! done in b65b58c89ec5b14e3980c47be531797a225b85ba